src/main/ruby/core/string.rb

```diff
@@ -1359,8 +1339,7 @@ def index(str, start=undefined)
     if str.kind_of? Regexp
       Truffle::Type.compatible_encoding self, str
 
-      m = Truffle::Mirror.reflect self
-      start = m.character_to_byte_index start
+      start = Truffle.invoke_primitive(:string_character_byte_index, self, start)
```

Comment:
I prefer with the "to" like `:string_character_to_byte_index` so it's clear it's conversion from what to what.
I.e., it takes an integer index as input, not a character like `string_character_index`.

Reply:
I see, that was the name of the primitive before as well.
I think it's a good occasion to rename it, but it doesn't have to be in this PR.

Reply:
I'm going to take a pass over the String primitives in another PR.